Add new `disallow_top_level_variables` rule (#4872)
realm/SwiftLint: CHANGELOG.md

* Trigger `prefer_self_in_static_references` rule on more type references.
  [SimplyDanny](https://github.com/SimplyDanny)
  

> * Add new `disallow_top_level_variables` rule that disallows declaring variables
>   and constants as the top-level declarations.

Reviewer: ```suggestion
* Add new `disallow_top_level_variables` rule that disallows variable and
  constant declarations on top level.  
```
The two spaces at the end are important to enforce a line break.
